raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -959,6 +972,10 @@ private FetchResponseData buildEmptyFetchResponse(
     ) {
         FetchRequestData request = (FetchRequestData) requestMetadata.data;
 
+        if (!hasValidClusterId(request)) {
```

Comment:
We only validate `FetchRequest` here? should we also add validation to the other 4 rpcs?